pages/register.tsx

```diff
@@ -0,0 +1,251 @@
+import { Inter } from 'next/font/google';
+import { useRouter } from 'next/router';
+import { FormEvent, HTMLAttributes, useState } from 'react';
+import { twMerge } from 'tailwind-merge';
+import { Button } from '~/components/Button';
+import { Input } from '~/components/Input';
+import { Modal } from '~/components/Modal';
+import { TitleText } from '~/components/TitleText';
+import { SvgCheckMarkIcon } from '~/icons/CheckMarkIcon';
+import { createUser } from '~/utils/apiRequests';
+import { handleUserError } from '~/utils/handleError';
+import { emailRegex, passwordRegex } from '~/utils/regexPatterns';
+
+const inter = Inter({ subsets: ['latin'] });
+
+export default function Login() {
+  const [firstName, setFirstName] = useState('');
+  const [firstNameErrorText, setFirstNameErrorText] = useState('');
+  const [isFirstNameError, setIsFirstNameError] = useState(false);
+
+  const [lastName, setLastName] = useState('');
+  const [lastNameErrorText, setLastNameErrorText] = useState('');
+  const [isLastNameError, setIsLastNameError] = useState(false);
+
+  const [email, setEmail] = useState('');
+  const [emailErrorText, setEmailErrorText] = useState('');
+  const [isEmailError, setIsEmailError] = useState(false);
+
+  const [password, setPassword] = useState('');
+  const [passwordErrorText, setPasswordErrorText] = useState('');
+  const [isPasswordError, setIsPasswordError] = useState(false);
+
+  const [isUserCreated, setIsUserCreated] = useState(false);
+
+  const router = useRouter();
+
+  async function handleRegister(e: FormEvent<HTMLFormElement>) {
+    try {
+      e.preventDefault();
+      if (!isAllFieldsValid()) return;
+      console.log('Handling registering!');
+      await createUser({
+        email: email,
+        firstName: firstName,
+        lastName: lastName,
+        password: password,
+      });
+      userCreatedSuccesfully();
+    } catch (e) {
+      handleUserError(e);
+    }
+  }
+
+  function userCreatedSuccesfully() {
+    setFirstName('');
+    setLastName('');
+    setEmail('');
+    setPassword('');
+    setIsUserCreated(true);
+    setTimeout(() => {
+      router.push('/').catch((e) => console.error(e));
+    }, 1000);
+  }
+
+  function clearAllErrors() {
+    setIsFirstNameError(false);
+    setIsLastNameError(false);
+    setIsEmailError(false);
+    setIsPasswordError(false);
+  }
+
+  function isAllFieldsValid(): boolean {
+    clearAllErrors();
+    let errorFound = false;
+    if (!isFirstNameValid()) errorFound = true;
+    if (!isLastNameValid()) errorFound = true;
+    if (!isEmailValid()) errorFound = true;
+    if (!isPasswordValid()) errorFound = true;
+
+    if (errorFound) {
+      return false;
+    }
+    return true;
```

Comment:
Tämän voi yksinkertaistaa
```suggestion
    return !errorFound;
```